### PR TITLE
[PHP] move count of array outside loop

### PR DESCRIPTION
**Pull request by acidvertigo**
In php precounting the size of array outside for loop increase performance because the count function is not executed every time.

### CODE REVIEW

**Comment by nadako** [comment]
Could someone who uses PHP target review this? @mockey maybe?

File: std/php/Boot.hx [hunk 1]
```diff
@@ -96,7 +96,8 @@ class _hx_array implements ArrayAccess, IteratorAggregate {
 	}
 
 	function remove($x) {
-		for($i = 0; $i < count($this->a); $i++)
+		$len = count($this->a);
```
Comment (RealyUniqueName) [inline std/php/Boot.hx]: I think this `count` is not necessary  because `$this->length` should always be equal to `count($this->a)` in this context.

**Comment by RealyUniqueName** [comment]
Looks fine to me. Except that one `count` in `remove()`. It can be completely optimized away :)

**Comment by mockey** [comment]
@RealyUniqueName You mean:`for($i = 0; $i < $this->length; $i++)` should be used instead?

There's also a `count` in `pop`:
https://github.com/HaxeFoundation/haxe/blob/development/std/php/Boot.hx#L89
which could be changed to: `$this->length--;` I think.

It's right that `count` inside the loop is not a good idea, but generally I think for simple loops like these `foreach` in PHP is preferable to C-style `for`, isn't it?

**Comment by RealyUniqueName** [comment]
I don't think there is significant difference between `for` and `foreach` performance wise. However `foreach` syntax is much cleaner and more convenient so i usually prefer it over `for`.

**Comment by mockey** [comment]
`foreach` should be slightly faster, but it doesn't matter much, that's right.
But for simple loops it should really be preferred.
@nadako Shall I create another PR then?

**Comment by nadako** [comment]
It's not necessary, you can add commits to this PR and then we merge+squash :)

**Comment by mockey** [comment]
> you can add commits to this PR

How that? It's not my PR.

**Comment by nadako** [comment]
Ooh, I'm sorry :) Yeah then, submit a new one.

**Comment by nadako** [comment]
replaced by #5471

**Pull request closed without merging**